### DISCUSSION TITLE
Executing moose's executable (on WSL) through Matlab.

### DISCUSSION THREAD

**Discussion by aaelmeli** [idaholab/moose#18707] [Q&A General] [answered]
Hi

I am using moose as the engine to solve the forward problem required for optimization problems. I am using Matlab's optimization toolbox for the optimization process (such as BFGS). Now, I am trying to call moose's executable from Matlab. I am using moose on WSL. 
So, is there a way to call moose's executable from Matlab? for instance,
something like this `system('cmd /C myprogram.exe');` 

Thank you.

**Comment by dschwen** [comment, marked as answer]
You need to use the `wsl` wrapper command. https://docs.microsoft.com/en-us/windows/wsl/wsl-config
